mlrun/feature_store/retrieval/base.py

```diff
@@ -102,18 +102,25 @@ def start(
             self.vector.save()
 
         for feature_set in feature_set_objects.values():
-            if not entity_timestamp_column and self._drop_indexes:
+            if self._drop_indexes:
                 self._append_drop_column(feature_set.spec.timestamp_key)
             for key in feature_set.spec.entities.keys():
                 self._append_index(key)
 
+        start_time = str_to_timestamp(start_time)
+        end_time = str_to_timestamp(end_time)
+        if start_time and not end_time:
+            # if end_time is not specified set it to now()
+            end_time = pd.Timestamp.now()
```

Comment:
Why setting it to `now()` here?  Doesn't the `_generate_vector` function know to ignore end time when not provided?

Reply:
Looks like all `_generate_vector` does with this info is pass it to `_get_engine_df` in which you handle the condition properly, so I think this should work without adding this `now()` to the query.